<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Jake Reed" />


<title>Using minimapR</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>
<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Using minimapR</h1>
<h4 class="author">Jake Reed</h4>



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>‘minimapR’ is a wrapper for ‘Minimap2’. ‘Minimap2’ is a very valuable long read aligner for the Pacbio and Oxford Nanopore Technologies sequencing platforms. ‘minimapR’ is an R wrapper for ‘minimap2’ which was developed by Heng Li <a href="mailto:me@liheng.org" class="email">me@liheng.org</a>. This tool aligns long reads to a reference genome and is used in many different bioinformatics workflows.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a><span class="kw">library</span>(minimapR)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a><span class="co">#&gt; Loading required package: Rsamtools</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true"></a><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true"></a><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true"></a><span class="co">#&gt; Attaching package: &#39;BiocGenerics&#39;</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true"></a><span class="co">#&gt; The following objects are masked from &#39;package:stats&#39;:</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true"></a><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true"></a><span class="co">#&gt; The following objects are masked from &#39;package:base&#39;:</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true"></a><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true"></a><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true"></a><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true"></a><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true"></a><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true"></a><span class="co">#&gt;     tapply, union, unique, unsplit, which.max, which.min</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true"></a><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true"></a><span class="co">#&gt; Loading required package: stats4</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true"></a><span class="co">#&gt; Attaching package: &#39;S4Vectors&#39;</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true"></a><span class="co">#&gt; The following object is masked from &#39;package:utils&#39;:</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true"></a><span class="co">#&gt;     findMatches</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true"></a><span class="co">#&gt; The following objects are masked from &#39;package:base&#39;:</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true"></a><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true"></a><span class="co">#&gt; Loading required package: IRanges</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true"></a><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true"></a><span class="co">#&gt; Loading required package: Biostrings</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true"></a><span class="co">#&gt; Loading required package: XVector</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true"></a><span class="co">#&gt; Attaching package: &#39;Biostrings&#39;</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true"></a><span class="co">#&gt; The following object is masked from &#39;package:base&#39;:</span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true"></a><span class="co">#&gt;     strsplit</span></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true"></a><span class="co">#&gt; Loading required package: git2r</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true"></a><span class="co">#&gt; Attaching package: &#39;git2r&#39;</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true"></a><span class="co">#&gt; The following objects are masked from &#39;package:Biostrings&#39;:</span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true"></a><span class="co">#&gt;     as.data.frame, head</span></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true"></a><span class="co">#&gt; The following object is masked from &#39;package:XVector&#39;:</span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true"></a><span class="co">#&gt;     as.data.frame</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true"></a><span class="co">#&gt; The following objects are masked from &#39;package:GenomicRanges&#39;:</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true"></a><span class="co">#&gt;     as.data.frame, merge</span></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true"></a><span class="co">#&gt; The following objects are masked from &#39;package:GenomeInfoDb&#39;:</span></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true"></a><span class="co">#&gt;     as.data.frame, merge</span></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true"></a><span class="co">#&gt; The following objects are masked from &#39;package:IRanges&#39;:</span></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true"></a><span class="co">#&gt;     as.data.frame, diff, merge</span></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true"></a><span class="co">#&gt; The following objects are masked from &#39;package:S4Vectors&#39;:</span></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true"></a><span class="co">#&gt;     as.data.frame, head, merge</span></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true"></a><span class="co">#&gt; The following object is masked from &#39;package:BiocGenerics&#39;:</span></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true"></a><span class="co">#&gt;     as.data.frame</span></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true"></a><span class="co">#&gt; Loading required package: pafr</span></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true"></a><span class="co">#&gt; Loading required package: ggplot2</span></span></code></pre></div>
</div>
<div id="installation" class="section level1">
<h1>Installation</h1>
<p>‘minimap2’ and ‘samtools’ must be installed on your system along with the R packages ‘Rsamtools’, ‘git2r’, and ‘pafr’. ‘minimap2’ can be installed on various operating systems by running the following function or following the instruction from the output of the function. “/your/path/to/directory/for/install” should be replaced with the path to the directory where you want to install ‘minimap2’.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true"></a><span class="kw">minimap2_installation</span>(<span class="st">&quot;/your/path/to/directory/for/install&quot;</span>)</span></code></pre></div>
<p>check if dependencies were successfully installed.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true"></a><span class="kw">minimap2_check</span>()</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true"></a><span class="co">#&gt; minimap2 is installed.</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true"></a><span class="co">#&gt;                                minimap2 </span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true"></a><span class="co">#&gt; &quot;/home/jake/software/minimap2/minimap2&quot;</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true"></a><span class="kw">samtools_check</span>()</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true"></a><span class="co">#&gt; samtools is installed.</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true"></a><span class="co">#&gt;                            samtools </span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true"></a><span class="co">#&gt; &quot;/home/jake/anaconda3/bin/samtools&quot;</span></span></code></pre></div>
</div>
<div id="usage" class="section level1">
<h1>Usage</h1>
<div id="downloading-data-for-example" class="section level2">
<h2>Downloading data for example</h2>
<p>We will download the reference genomes and the query sequences for the example. The reference genomes are the human genome (GRCh38) and the yeast genome (S288C). The query sequences are the human ONT reads and yeast HIFI reads.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true"></a>tmp_folder &lt;-<span class="st"> </span><span class="kw">tempdir</span>()</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true"></a><span class="kw">cat</span>(<span class="st">&quot;Temporary folder is:&quot;</span>, tmp_folder, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true"></a><span class="co">#&gt; Temporary folder is: /tmp/RtmpJqcozw</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true"></a>href_url &lt;-<span class="st"> &quot;https://github.com/jake-bioinfo/minimapR/raw/master/inst/extdata/GRCh38_chr1_50m.fa&quot;</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true"></a>hfq_url &lt;-<span class="st"> &quot;https://github.com/jake-bioinfo/minimapR/raw/master/inst/extdata/ont_hs_sample.fastq.gz&quot;</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true"></a>yref_url &lt;-<span class="st"> &quot;https://github.com/jake-bioinfo/minimapR/raw/master/inst/extdata/S288C_ref_genome.fasta&quot;</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true"></a>yfq_url &lt;-<span class="st"> &quot;https://github.com/jake-bioinfo/minimapR/raw/master/inst/extdata/yeast_sample_hifi.fastq.gz&quot;</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true"></a>url_list &lt;-<span class="st"> </span><span class="kw">c</span>(href_url, hfq_url, yref_url, yfq_url)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true"></a><span class="kw">lapply</span>(url_list, <span class="cf">function</span>(x) <span class="kw">download.file</span>(x, <span class="dt">destfile =</span> <span class="kw">file.path</span>(tmp_folder, <span class="kw">basename</span>(x))))</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true"></a><span class="co">#&gt; [[1]]</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true"></a><span class="co">#&gt; [1] 0</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true"></a><span class="co">#&gt; [[2]]</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true"></a><span class="co">#&gt; [1] 0</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true"></a><span class="co">#&gt; [[3]]</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true"></a><span class="co">#&gt; [1] 0</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true"></a><span class="co">#&gt; [[4]]</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true"></a><span class="co">#&gt; [1] 0</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true"></a></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true"></a><span class="co"># Contents of the temporary folder</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true"></a><span class="kw">cat</span>(<span class="st">&quot;Contents of the temporary folder are:&quot;</span>, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true"></a><span class="co">#&gt; Contents of the temporary folder are:</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true"></a>fa_list &lt;-<span class="st"> </span><span class="kw">list.files</span>(tmp_folder, <span class="dt">pattern =</span> <span class="st">&quot;.fa&quot;</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>)</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true"></a>fa_list</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true"></a><span class="co">#&gt; [1] &quot;/tmp/RtmpJqcozw/GRCh38_chr1_50m.fa&quot;        </span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true"></a><span class="co">#&gt; [2] &quot;/tmp/RtmpJqcozw/ont_hs_sample.fastq.gz&quot;    </span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true"></a><span class="co">#&gt; [3] &quot;/tmp/RtmpJqcozw/S288C_ref_genome.fasta&quot;    </span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true"></a><span class="co">#&gt; [4] &quot;/tmp/RtmpJqcozw/yeast_sample_hifi.fastq.gz&quot;</span></span></code></pre></div>
</div>
<div id="aligning-reads-to-the-reference-genomes" class="section level2">
<h2>Aligning reads to the reference genomes</h2>
<p>We will align the human ONT reads to the human genome and the yeast HIFI reads to the yeast genome using ‘minimap2’. The ‘minimap2’ function returns a data frame with the alignment information.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true"></a><span class="co"># Human ONT alignment</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true"></a>h_out &lt;-<span class="st"> </span><span class="kw">file.path</span>(tmp_folder, <span class="st">&quot;ont_hs_sample&quot;</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true"></a>h_alignment &lt;-<span class="st"> </span><span class="kw">minimap2</span>(<span class="dt">reference =</span> fa_list[<span class="dv">1</span>], </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true"></a>                        <span class="dt">query_sequences =</span> fa_list[<span class="dv">2</span>], </span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true"></a>                        <span class="dt">output_file_prefix =</span> h_out, </span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true"></a>                        <span class="dt">preset_string =</span> <span class="st">&quot;map-ont&quot;</span>, </span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true"></a>                        <span class="dt">threads =</span> <span class="dv">4</span>, </span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true"></a>                        <span class="dt">return =</span> <span class="ot">TRUE</span>, </span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true"></a>                        <span class="dt">verbose =</span> <span class="ot">TRUE</span>)</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true"></a><span class="co">#&gt; Generating output file:  /tmp/RtmpJqcozw/ont_hs_sample</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true"></a><span class="co">#&gt; Running minimap2...</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true"></a><span class="co">#&gt; Running minimap with the following command:</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true"></a><span class="co">#&gt; /home/jake/software/minimap2/minimap2 -ax map-ont -t 4 /tmp/RtmpJqcozw/GRCh38_chr1_50m.fa /tmp/RtmpJqcozw/ont_hs_sample.fastq.gz -o /tmp/RtmpJqcozw/ont_hs_sample.sam</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true"></a></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true"></a><span class="co"># Yeast HIFI alignment</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true"></a>y_out &lt;-<span class="st"> </span><span class="kw">file.path</span>(tmp_folder, <span class="st">&quot;yeast_sample_hifi&quot;</span>)</span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true"></a>y_alignment &lt;-<span class="st"> </span><span class="kw">minimap2</span>(<span class="dt">reference =</span> fa_list[<span class="dv">3</span>], </span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true"></a>                        <span class="dt">query_sequences =</span> fa_list[<span class="dv">4</span>], </span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true"></a>                        <span class="dt">output_file_prefix =</span> y_out, </span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true"></a>                        <span class="dt">preset_string =</span> <span class="st">&quot;map-hifi&quot;</span>, </span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true"></a>                        <span class="dt">threads =</span> <span class="dv">4</span>, </span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true"></a>                        <span class="dt">return =</span> <span class="ot">TRUE</span>, </span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true"></a>                        <span class="dt">verbose =</span> <span class="ot">TRUE</span>)</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true"></a><span class="co">#&gt; Generating output file:  /tmp/RtmpJqcozw/yeast_sample_hifi</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true"></a><span class="co">#&gt; Running minimap2...</span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true"></a><span class="co">#&gt; Running minimap with the following command:</span></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true"></a><span class="co">#&gt; /home/jake/software/minimap2/minimap2 -ax map-hifi -t 4 /tmp/RtmpJqcozw/S288C_ref_genome.fasta /tmp/RtmpJqcozw/yeast_sample_hifi.fastq.gz -o /tmp/RtmpJqcozw/yeast_sample_hifi.sam</span></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true"></a></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true"></a><span class="co"># Check the alignment</span></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true"></a><span class="kw">cat</span>(<span class="st">&quot;Alignment for the human sample is:</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true"></a><span class="co">#&gt; Alignment for the human sample is:</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true"></a>h_alignment[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">7</span>]</span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true"></a><span class="co">#&gt;              qname flag                   rname strand   pos qwidth mapq</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true"></a><span class="co">#&gt; 1 SRR24462105.6945 2064 NC_000001.11_1-50000000      - 10001    588    3</span></span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true"></a><span class="co">#&gt; 2 SRR24462105.6945 2064 NC_000001.11_1-50000000      - 10001    457    1</span></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true"></a><span class="co">#&gt; 3 SRR24462105.6945 2064 NC_000001.11_1-50000000      - 10001    476    1</span></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true"></a><span class="co">#&gt; 4 SRR24462105.6945 2064 NC_000001.11_1-50000000      - 10001    428    1</span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true"></a><span class="co">#&gt; 5 SRR24462105.6945 2064 NC_000001.11_1-50000000      - 10001    527   52</span></span>
<span id="cb5-39"><a href="#cb5-39" aria-hidden="true"></a><span class="kw">cat</span>(<span class="st">&quot;Alignment for the yeast sample is:</span><span class="ch">\n</span><span class="st">&quot;</span>)</span>
<span id="cb5-40"><a href="#cb5-40" aria-hidden="true"></a><span class="co">#&gt; Alignment for the yeast sample is:</span></span>
<span id="cb5-41"><a href="#cb5-41" aria-hidden="true"></a>y_alignment[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">7</span>]</span>
<span id="cb5-42"><a href="#cb5-42" aria-hidden="true"></a><span class="co">#&gt;               qname flag          rname strand pos qwidth mapq</span></span>
<span id="cb5-43"><a href="#cb5-43" aria-hidden="true"></a><span class="co">#&gt; 1 SRR13577847.41206    0 ref|NC_001133|      +   1   8855   60</span></span>
<span id="cb5-44"><a href="#cb5-44" aria-hidden="true"></a><span class="co">#&gt; 2 SRR13577847.14254   16 ref|NC_001133|      -   1   8966   60</span></span>
<span id="cb5-45"><a href="#cb5-45" aria-hidden="true"></a><span class="co">#&gt; 3 SRR13577847.18785   16 ref|NC_001133|      -   1   8420   60</span></span>
<span id="cb5-46"><a href="#cb5-46" aria-hidden="true"></a><span class="co">#&gt; 4 SRR13577847.31750   16 ref|NC_001133|      -   1   9070   60</span></span>
<span id="cb5-47"><a href="#cb5-47" aria-hidden="true"></a><span class="co">#&gt; 5 SRR13577847.40833   16 ref|NC_001133|      -   1   8704   60</span></span></code></pre></div>
</div>
</div>
<div id="clean-up" class="section level1">
<h1>Clean up</h1>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true"></a><span class="co"># Clean up</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true"></a><span class="kw">unlink</span>(tmp_folder, <span class="dt">recursive =</span> <span class="ot">TRUE</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true"></a><span class="kw">sessionInfo</span>()</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true"></a><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true"></a><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true"></a><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true"></a><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 </span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true"></a><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true"></a><span class="co">#&gt; locale:</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true"></a><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true"></a><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true"></a><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true"></a><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true"></a><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true"></a><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true"></a><span class="co">#&gt; time zone: America/New_York</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true"></a><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true"></a><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true"></a><span class="co">#&gt; [8] base     </span></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true"></a><span class="co">#&gt; other attached packages:</span></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true"></a><span class="co">#&gt;  [1] minimapR_0.0.1.0     pafr_0.0.2           ggplot2_3.5.1       </span></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true"></a><span class="co">#&gt;  [4] git2r_0.33.0         Rsamtools_2.20.0     Biostrings_2.72.0   </span></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true"></a><span class="co">#&gt;  [7] XVector_0.44.0       GenomicRanges_1.56.0 GenomeInfoDb_1.40.0 </span></span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true"></a><span class="co">#&gt; [10] IRanges_2.38.0       S4Vectors_0.42.0     BiocGenerics_0.50.0 </span></span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true"></a><span class="co">#&gt; </span></span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true"></a><span class="co">#&gt;  [1] sass_0.4.9              utf8_1.2.4              generics_0.1.3         </span></span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true"></a><span class="co">#&gt;  [4] bitops_1.0-7            stringi_1.8.4           digest_0.6.36          </span></span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true"></a><span class="co">#&gt;  [7] magrittr_2.0.3          evaluate_0.24.0         grid_4.4.1             </span></span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true"></a><span class="co">#&gt; [10] fastmap_1.2.0           jsonlite_1.8.8          httr_1.4.7             </span></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true"></a><span class="co">#&gt; [13] fansi_1.0.6             UCSC.utils_1.0.0        scales_1.3.0           </span></span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true"></a><span class="co">#&gt; [16] codetools_0.2-20        jquerylib_0.1.4         cli_3.6.3              </span></span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true"></a><span class="co">#&gt; [19] rlang_1.1.4             crayon_1.5.3            munsell_0.5.1          </span></span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true"></a><span class="co">#&gt; [22] withr_3.0.0             cachem_1.1.0            yaml_2.3.9             </span></span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true"></a><span class="co">#&gt; [25] tools_4.4.1             parallel_4.4.1          BiocParallel_1.38.0    </span></span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true"></a><span class="co">#&gt; [28] dplyr_1.1.4             colorspace_2.1-0        GenomeInfoDbData_1.2.12</span></span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true"></a><span class="co">#&gt; [31] vctrs_0.6.5             R6_2.5.1                lifecycle_1.0.4        </span></span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true"></a><span class="co">#&gt; [34] zlibbioc_1.50.0         stringr_1.5.1           pkgconfig_2.0.3        </span></span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true"></a><span class="co">#&gt; [37] bslib_0.7.0             pillar_1.9.0            gtable_0.3.5           </span></span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true"></a><span class="co">#&gt; [40] glue_1.7.0              xfun_0.45               tibble_3.2.1           </span></span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true"></a><span class="co">#&gt; [43] tidyselect_1.2.1        knitr_1.48              htmltools_0.5.8.1      </span></span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true"></a><span class="co">#&gt; [46] rmarkdown_2.27          compiler_4.4.1</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
