<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using minimapR • minimapR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using minimapR">
<meta property="og:description" content="minimapR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">minimapR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/minimapR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jake-bioinfo/minimapR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="minimapR_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Using minimapR</h1>
                        <h4 data-toc-skip class="author">Jake Reed</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jake-bioinfo/minimapR/blob/HEAD/vignettes/minimapR.Rmd" class="external-link"><code>vignettes/minimapR.Rmd</code></a></small>
      <div class="hidden name"><code>minimapR.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>‘minimapR’ is a wrapper for ‘Minimap2’. ‘Minimap2’ is a very valuable long read aligner for the Pacbio and Oxford Nanopore Technologies sequencing platforms. ‘minimapR’ is an R wrapper for ‘minimap2’ which was developed by Heng Li <a href="mailto:me@liheng.org" class="email">me@liheng.org</a>. This tool aligns long reads to a reference genome and is used in many different bioinformatics workflows.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jake-bioinfo/minimapR" class="external-link">minimapR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: Rsamtools</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,</span></span>
<span><span class="co">#&gt;     tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: Biostrings</span></span>
<span><span class="co">#&gt; Loading required package: XVector</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'Biostrings'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     strsplit</span></span>
<span><span class="co">#&gt; Loading required package: git2r</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'git2r'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:Biostrings':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     as.data.frame, head</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:XVector':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     as.data.frame</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:GenomicRanges':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     as.data.frame, merge</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:GenomeInfoDb':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     as.data.frame, merge</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:IRanges':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     as.data.frame, diff, merge</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:S4Vectors':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     as.data.frame, head, merge</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:BiocGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     as.data.frame</span></span>
<span><span class="co">#&gt; Loading required package: pafr</span></span>
<span><span class="co">#&gt; Loading required package: ggplot2</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>‘minimap2’ and ‘samtools’ must be installed on your system along with the R packages ‘Rsamtools’, ‘git2r’, and ‘pafr’. ‘minimap2’ can be installed on various operating systems by running the following function or following the instruction from the output of the function. “/your/path/to/directory/for/install” should be replaced with the path to the directory where you want to install ‘minimap2’.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/minimap2_installation.html">minimap2_installation</a></span><span class="op">(</span><span class="st">"/your/path/to/directory/for/install"</span><span class="op">)</span></span></code></pre></div>
<p>check if dependencies were successfully installed.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/minimap2_check.html">minimap2_check</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; minimap2 is installed.</span></span>
<span><span class="co">#&gt;                                minimap2 </span></span>
<span><span class="co">#&gt; "/home/jake/software/minimap2/minimap2"</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/samtools_check.html">samtools_check</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; samtools is installed.</span></span>
<span><span class="co">#&gt;                            samtools </span></span>
<span><span class="co">#&gt; "/home/jake/anaconda3/bin/samtools"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<div class="section level3">
<h3 id="downloading-data-for-example">Downloading data for example<a class="anchor" aria-label="anchor" href="#downloading-data-for-example"></a>
</h3>
<p>We will download the reference genomes and the query sequences for the example. The reference genomes are the human genome (GRCh38) and the yeast genome (S288C). The query sequences are the human ONT reads and yeast HIFI reads.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp_folder</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Temporary folder is:"</span>, <span class="va">tmp_folder</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Temporary folder is: /tmp/RtmplRRS63</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">href_url</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/jake-bioinfo/minimapR/raw/master/inst/extdata/GRCh38_chr1_50m.fa"</span></span>
<span><span class="va">hfq_url</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/jake-bioinfo/minimapR/raw/master/inst/extdata/ont_hs_sample.fastq.gz"</span></span>
<span><span class="va">yref_url</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/jake-bioinfo/minimapR/raw/master/inst/extdata/S288C_ref_genome.fasta"</span></span>
<span><span class="va">yfq_url</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/jake-bioinfo/minimapR/raw/master/inst/extdata/yeast_sample_hifi.fastq.gz"</span></span>
<span><span class="va">url_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">href_url</span>, <span class="va">hfq_url</span>, <span class="va">yref_url</span>, <span class="va">yfq_url</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">url_list</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">x</span>, destfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp_folder</span>, <span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [[1]]</span></span>
<span><span class="co">#&gt; [1] 0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[2]]</span></span>
<span><span class="co">#&gt; [1] 0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[3]]</span></span>
<span><span class="co">#&gt; [1] 0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; [[4]]</span></span>
<span><span class="co">#&gt; [1] 0</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Contents of the temporary folder</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Contents of the temporary folder are:"</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Contents of the temporary folder are:</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fa_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">tmp_folder</span>, pattern <span class="op">=</span> <span class="st">".fa"</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">fa_list</span></span>
<span><span class="co">#&gt; [1] "/tmp/RtmplRRS63/GRCh38_chr1_50m.fa"        </span></span>
<span><span class="co">#&gt; [2] "/tmp/RtmplRRS63/ont_hs_sample.fastq.gz"    </span></span>
<span><span class="co">#&gt; [3] "/tmp/RtmplRRS63/S288C_ref_genome.fasta"    </span></span>
<span><span class="co">#&gt; [4] "/tmp/RtmplRRS63/yeast_sample_hifi.fastq.gz"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="aligning-reads-to-the-reference-genomes">Aligning reads to the reference genomes<a class="anchor" aria-label="anchor" href="#aligning-reads-to-the-reference-genomes"></a>
</h3>
<p>We will align the human ONT reads to the human genome and the yeast HIFI reads to the yeast genome using ‘minimap2’. The ‘minimap2’ function returns a data frame with the alignment information.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Human ONT alignment</span></span>
<span><span class="va">h_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp_folder</span>, <span class="st">"ont_hs_sample"</span><span class="op">)</span></span>
<span><span class="va">h_alignment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/minimap2.html">minimap2</a></span><span class="op">(</span>reference <span class="op">=</span> <span class="va">fa_list</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, </span>
<span>                        query_sequences <span class="op">=</span> <span class="va">fa_list</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, </span>
<span>                        output_file_prefix <span class="op">=</span> <span class="va">h_out</span>, </span>
<span>                        preset_string <span class="op">=</span> <span class="st">"map-ont"</span>, </span>
<span>                        threads <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                        return <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                        verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Generating output file:  /tmp/RtmplRRS63/ont_hs_sample</span></span>
<span><span class="co">#&gt; Running minimap2...</span></span>
<span><span class="co">#&gt; Running minimap with the following command:</span></span>
<span><span class="co">#&gt; /home/jake/software/minimap2/minimap2 -ax map-ont -t 4 /tmp/RtmplRRS63/GRCh38_chr1_50m.fa /tmp/RtmplRRS63/ont_hs_sample.fastq.gz -o /tmp/RtmplRRS63/ont_hs_sample.sam</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Yeast HIFI alignment</span></span>
<span><span class="va">y_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tmp_folder</span>, <span class="st">"yeast_sample_hifi"</span><span class="op">)</span></span>
<span><span class="va">y_alignment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/minimap2.html">minimap2</a></span><span class="op">(</span>reference <span class="op">=</span> <span class="va">fa_list</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>, </span>
<span>                        query_sequences <span class="op">=</span> <span class="va">fa_list</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span>, </span>
<span>                        output_file_prefix <span class="op">=</span> <span class="va">y_out</span>, </span>
<span>                        preset_string <span class="op">=</span> <span class="st">"map-hifi"</span>, </span>
<span>                        threads <span class="op">=</span> <span class="fl">4</span>, </span>
<span>                        return <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                        verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Generating output file:  /tmp/RtmplRRS63/yeast_sample_hifi</span></span>
<span><span class="co">#&gt; Running minimap2...</span></span>
<span><span class="co">#&gt; Running minimap with the following command:</span></span>
<span><span class="co">#&gt; /home/jake/software/minimap2/minimap2 -ax map-hifi -t 4 /tmp/RtmplRRS63/S288C_ref_genome.fasta /tmp/RtmplRRS63/yeast_sample_hifi.fastq.gz -o /tmp/RtmplRRS63/yeast_sample_hifi.sam</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Check the alignment</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Alignment for the human sample is:\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Alignment for the human sample is:</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">h_alignment</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span></span>
<span><span class="co">#&gt;              qname flag                   rname strand   pos qwidth mapq</span></span>
<span><span class="co">#&gt; 1 SRR24462105.6945 2064 NC_000001.11_1-50000000      - 10001    588    3</span></span>
<span><span class="co">#&gt; 2 SRR24462105.6945 2064 NC_000001.11_1-50000000      - 10001    457    1</span></span>
<span><span class="co">#&gt; 3 SRR24462105.6945 2064 NC_000001.11_1-50000000      - 10001    476    1</span></span>
<span><span class="co">#&gt; 4 SRR24462105.6945 2064 NC_000001.11_1-50000000      - 10001    428    1</span></span>
<span><span class="co">#&gt; 5 SRR24462105.6945 2064 NC_000001.11_1-50000000      - 10001    527   52</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Alignment for the yeast sample is:\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Alignment for the yeast sample is:</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">y_alignment</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span></span>
<span><span class="co">#&gt;               qname flag          rname strand pos qwidth mapq</span></span>
<span><span class="co">#&gt; 1 SRR13577847.41206    0 ref|NC_001133|      +   1   8855   60</span></span>
<span><span class="co">#&gt; 2 SRR13577847.14254   16 ref|NC_001133|      -   1   8966   60</span></span>
<span><span class="co">#&gt; 3 SRR13577847.18785   16 ref|NC_001133|      -   1   8420   60</span></span>
<span><span class="co">#&gt; 4 SRR13577847.31750   16 ref|NC_001133|      -   1   9070   60</span></span>
<span><span class="co">#&gt; 5 SRR13577847.40833   16 ref|NC_001133|      -   1   8704   60</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="clean-up">Clean up<a class="anchor" aria-label="anchor" href="#clean-up"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Clean up</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">tmp_folder</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span>list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ls.html" class="external-link">ls</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/New_York</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] minimapR_0.0.1.0     pafr_0.0.2           ggplot2_3.4.3       </span></span>
<span><span class="co">#&gt;  [4] git2r_0.33.0         Rsamtools_2.20.0     Biostrings_2.72.0   </span></span>
<span><span class="co">#&gt;  [7] XVector_0.44.0       GenomicRanges_1.56.0 GenomeInfoDb_1.40.0 </span></span>
<span><span class="co">#&gt; [10] IRanges_2.38.0       S4Vectors_0.42.0     BiocGenerics_0.50.0 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] generics_0.1.3          utf8_1.2.4              sass_0.4.9             </span></span>
<span><span class="co">#&gt;  [4] bitops_1.0-7            stringi_1.8.4           digest_0.6.36          </span></span>
<span><span class="co">#&gt;  [7] magrittr_2.0.3          evaluate_0.24.0         grid_4.4.1             </span></span>
<span><span class="co">#&gt; [10] fastmap_1.2.0           jsonlite_1.8.8          httr_1.4.7             </span></span>
<span><span class="co">#&gt; [13] purrr_1.0.2             fansi_1.0.6             UCSC.utils_1.0.0       </span></span>
<span><span class="co">#&gt; [16] scales_1.3.0            codetools_0.2-20        textshaping_0.4.0      </span></span>
<span><span class="co">#&gt; [19] jquerylib_0.1.4         cli_3.6.3               rlang_1.1.4            </span></span>
<span><span class="co">#&gt; [22] crayon_1.5.3            munsell_0.5.1           withr_3.0.0            </span></span>
<span><span class="co">#&gt; [25] cachem_1.1.0            yaml_2.3.8              tools_4.4.1            </span></span>
<span><span class="co">#&gt; [28] parallel_4.4.1          BiocParallel_1.38.0     memoise_2.0.1          </span></span>
<span><span class="co">#&gt; [31] dplyr_1.1.4             colorspace_2.1-0        GenomeInfoDbData_1.2.12</span></span>
<span><span class="co">#&gt; [34] vctrs_0.6.5             R6_2.5.1                lifecycle_1.0.4        </span></span>
<span><span class="co">#&gt; [37] stringr_1.5.1           zlibbioc_1.50.0         fs_1.6.4               </span></span>
<span><span class="co">#&gt; [40] htmlwidgets_1.6.4       ragg_1.3.2              pkgconfig_2.0.3        </span></span>
<span><span class="co">#&gt; [43] desc_1.4.3              pkgdown_2.0.9           bslib_0.7.0            </span></span>
<span><span class="co">#&gt; [46] pillar_1.9.0            gtable_0.3.5            glue_1.7.0             </span></span>
<span><span class="co">#&gt; [49] systemfonts_1.1.0       tidyselect_1.2.1        xfun_0.45              </span></span>
<span><span class="co">#&gt; [52] tibble_3.2.1            knitr_1.47              htmltools_0.5.8.1      </span></span>
<span><span class="co">#&gt; [55] rmarkdown_2.27          compiler_4.4.1</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Jake Reed, Pradyun Pulipaka.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
